### PR TITLE
feat: add promise queue package

### PR DESCRIPTION
**Pull request by tlebon**
This adds a PromiseQueue util that can be used in the core to queue sending of messages

### CODE REVIEW

File: packages/promise-queue/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
```
Comment (tlebon) [inline packages/promise-queue/jest.config.js:1]: the babel config has a copyright and this doesn't. not sure if either of these type of files need one.

**Comment by tlebon** [comment]
fine for me but i cant approve really.

File: packages/promise-queue/src/PromiseQueue.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  paused: false,
+};
+
+export class PromiseQueue {
+  private blocked: boolean;
+  private runningTasks: number;
+  private interval?: number;
+  private paused: boolean;
+  private readonly concurrent: number;
+  private readonly logger?: {warn: (...args: any[]) => void};
```
Comment (tlebon) [inline packages/promise-queue/src/PromiseQueue.ts:35]: just curious if we should include a default logger option as well as allowing the consumer to set one.

File: packages/promise-queue/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    [`${projectName}.test`]: `${__dirname}/src/test/browser/index.js`,
+  },
+  mode: 'production',
+  output: {
+    filename: '[name].bundle.js',
+    library: 'PromiseQueue',
+    path: `${__dirname}/dist`,
+    publicPath: '/',
+  },
+};
```
Comment (atomrc) [inline packages/promise-queue/webpack.config.js:37]: Oh, we don't need this :D

File: packages/promise-queue/src/PromiseQueue.ts (hunk 2 above)
Comment (atomrc) [reply packages/promise-queue/src/PromiseQueue.ts:35]: I'd like our libraries to be rather agnostic from any particular logger. 
Initially it pulled the entire `logdown` lib for logging, but I don't think it makes sense. If at some point we migrate the webapp's logger, I would like the change to reflect on all the libraries we use. 
So I think injecting the logger makes a lot of sense and let the webapp completely control how things are logged and with what namespace

**Pull request merged**